lib/vector-buffers/src/lib.rs

```diff
@@ -84,38 +84,28 @@ impl Arbitrary for WhenFull {
     }
 }
 
-/// An item that can be buffered.
+/// An item that can be buffered in memory.
 ///
-/// This supertrait serves as the base trait for any item that can be pushed into a buffer.
-pub trait Bufferable:
-    AddBatchNotifier
-    + ByteSizeOf
-    + Encodable
-    + EventCount
-    + Debug
-    + Send
-    + Sync
-    + Unpin
-    + Sized
-    + 'static
+/// This supertrait serves as the base trait for any item that can be pushed into a memory buffer.
+pub trait InMemoryBufferable:
+    AddBatchNotifier + ByteSizeOf + EventCount + Debug + Send + Sync + Unpin + Sized + 'static
 {
 }
```

Comment:
The issue I encountered is that we need to pass an `Instant` through vector's `limited_queue` in `SourceSender`, and an `Instant` is not `Encodable` from my understanding. This is so that we can measure the time from when the payload is sent to from the source to the time when the payload is enqueued on the buffer.

Reply:
I can add a comment explaining why it exists.

Reply:
No need, makes sense: less of a "this code is confusing" thing and more of a "trying to understand why the change was made" thing.